File: 프로그래머스/2/131127. 할인 행사/할인 행사.js
function solution(want, number, discount) {
    var answer = 0;
    
    let Check = ((arr) => {
        var tmp = new Map();
        arr.forEach((a) => tmp.set(a, (tmp.get(a) || 0) + 1));
        for(let i=0; i<want.length; i++){
            if(tmp.get(want[i]) !== number[i])
                return false;
        }
        
        return true;
    });
  
    for(let i=0; i<=discount.length-10; i++){
      var check = discount.slice(i, i+10);
      if(Check(check))
          answer++;
    }
    
    return answer;
}

File: 프로그래머스/2/131127. 할인 행사/할인 행사.cpp
#include <string>
#include <vector>
#include <map>

using namespace std;

int solution(vector<string> want, vector<int> number, vector<string> discount) {
    int answer = 0;
    
    map<string, int> cnt;
    for(int i=0; i<want.size(); i++)
        cnt[want[i]] = number[i];
 
    for(int i=0; i<=discount.size()-10; i++){
        map<string, int> check;
        for(int j=i; j<i+10; j++)
            check[discount[j]]++;

        if(check == cnt)
            answer++;
    }
    
    return answer;
}